### PR TITLE
Bude 1069 1 add truncating ellipses to header columns

### PR DESCRIPTION
**Pull request by NYConfidential**
## PR Checklist

Storybook can be viewed [here](https://docspot.adnxs.net/projects/lucid/BUDE-1069_1_add_truncating_ellipses_to_header_columns)

- Manually tested across supported browsers

  - [ x] Chrome
  - [] Firefox
  - [x ] Safari

- [x ] Unit tests written (`common` at minimum)
- [x ] PR has one of the `semver-` labels
- [x] Two core team engineer approvals
- [x] One core team UX approval
- [ ] 
image

### CODE REVIEW

File: src/components/DataTable/DataTable.tsx [hunk 1]
```diff
@@ -428,14 +439,14 @@ export const DataTable = (props: IDataTableProps) => {
 													!_.isNil(columnProps.hasBorderRight)
 														? columnProps.hasBorderRight
 														: isFixedColumn &&
-														columnIndex + 1 + (isSelectable ? 1 : 0) ===
-														endColumn
+														  columnIndex + 1 + (isSelectable ? 1 : 0) ===
+																endColumn
```
Comment (scotthollandjr) [inline src/components/DataTable/DataTable.tsx:443]: i'm worried about this indentation. i assume it was auto-indented but still looks a little weird

File: src/components/Table/Table.less [hunk 2]
```diff
@@ -209,22 +209,42 @@
 				}
 			}
 
+
+			// header cell content displays ellipses if text overflows
+			// and <DataTable hasFixedHeader={true} truncateContent={true}>
+			&.@{prefix}-Table-truncate-content {
+				& > .@{prefix}-Table-Th-inner {
+					//display: flex;
```
Comment (scotthollandjr) [inline src/components/Table/Table.less:217]: unnecessary comment

**Review by scotthollandjr — APPROVED** [review]

**Review by ginoafable-xandr — APPROVED** [review]
Approved - Pending: change the word "Occupation" to something like "Daytime Occupation" or make the column narrower to show "Occupati..." to show a better example. Other users might not know why the elipses is there after the whole word was shown if they don't open the code and find out the column is actually "Occupation Occupation Occupation" which is not a common header title.
Screen Shot 2021-04-08 at 3 01 50 PM

File: src/components/DataTable/DataTable.tsx (hunk 1 above)
Comment (NYConfidential) [reply src/components/DataTable/DataTable.tsx:443]: Yes, it's the linter... But should be fine

**Review by lancempoe — CHANGES_REQUESTED** [review]
Please keep all PRs only related to the task at hand. There are all kinds of changes here that make it really hard to review. 

If there was a linting change recently, then these files need updated PRIOR to this change.

**Review by lancempoe — APPROVED** [review]

**Pull request merged**